include/context.h

```diff
@@ -215,6 +226,78 @@ class Context
         {
             return m_map.remove(key);
         }
+
+        bool createGroup(const std::string& gname)
+        {
+            return m_map.createGroup(gname);
+        }
+
+        bool hasGroup(const std::string& gname)
+        {
+            return getGroup(gname) != nullptr;
+        }
+
+        bool deleteGroup(const std::string& gname)
+        {
+            return m_map.deleteGroup(gname);
+        }
+
+        bool groupAddKey(const std::string& gname, const std::string& key)
+        {
+            GroupPtr gptr = getGroup(gname);
+            if(!gptr) return false;
+            return gptr->add(key);
+        }
+
+        bool groupHasKey(const std::string& gname, const std::string& key)
+        {
+            GroupPtr gptr = getGroup(gname);
+            if(!gptr) return false;
+            return gptr->has(key);
+        }
+
+        bool groupRemoveKey(const std::string& gname, const std::string& key)
+        {
+            GroupPtr gptr = getGroup(gname);
+            if(!gptr) return false;
+            return gptr->remove(key);
+        }
+
+        template<typename T>
+        bool groupSet(const std::string& gname, const std::string& key, T&& val, std::chrono::seconds ttl = std::chrono::seconds(0), GlobalContextMap::OnExpired onExpired = nullptr)
```

Comment:
after some thinking I lean to removal of groupSet/Get interfaces: they provide no real advantage in performance (std:map's tree vs our hash table) while likely confusing users

Reply:
Comments added